src/main/java/filter/SemanticException.java

```diff
@@ -0,0 +1,14 @@
+package filter;
+
+import filter.expression.QualifierType;
+
+@SuppressWarnings("serial")
+public class SemanticException extends FilterException {
+    
+    public static final String ERR_MSG = "\"%s\" expects %s";
```

Comment:
I know this is more or less unambiguous in context, but we should try not to abbreviate names, just as a matter of style.